gcloud-java-compute/src/main/java/com/google/gcloud/compute/DeprecationStatus.java

```diff
@@ -64,12 +67,77 @@
     DELETED
   }
 
-  DeprecationStatus(Long deleted, Long deprecated, Long obsolete, T replacement, Status status) {
-    this.deleted = deleted;
-    this.deprecated = deprecated;
-    this.obsolete = obsolete;
-    this.replacement = replacement;
-    this.status = status;
+  /**
+   * A builder for {@code DeprecationStatus} objects.
+   *
+   * @param <T> The Google Compute Engine resource identity to which the deprecation status refers
+   *     to
+   */
+  public static final class Builder<T extends ResourceId> {
+
+    private Long deleted;
+    private Long deprecated;
+    private Long obsolete;
+    private T replacement;
+    private Status status;
+
+    Builder() {}
+
+    /**
+     * Sets the timestamp on or after which the deprecation state of this resource will be changed
+     * to {@link Status#DELETED}. In milliseconds since epoch.
+     */
+    public Builder<T> deleted(Long deleted) {
+      this.deleted = deleted;
+      return this;
+    }
+
+    /**
+     * Sets the timestamp on or after which the deprecation state of this resource will be changed
+     * to {@link Status#DEPRECATED}. In milliseconds since epoch.
+     */
+    public Builder<T> deprecated(Long deprecated) {
+      this.deprecated = deprecated;
+      return this;
+    }
+
+    /**
+     * Sets the timestamp on or after which the deprecation state of this resource will be changed
+     * to {@link Status#OBSOLETE}. In milliseconds since epoch.
+     */
+    public Builder<T> obsolete(Long obsolete) {
+      this.obsolete = obsolete;
+      return this;
+    }
+
+    /**
+     * Sets the identity of the suggested replacement for a deprecated resource. The suggested
+     * replacement resource must be the same kind of resource as the deprecated resource.
+     */
+    public Builder<T> replacement(T replacement) {
+      this.replacement = replacement;
+      return this;
+    }
+
+    public Builder<T> status(Status status) {
```

Comment:
Actually, I see it's supposed to be public.  We could add javadoc to this method.